### PR TITLE
Better error messages for numerical solver

### PR DESCRIPTION
**Pull request by tanayag**
@asmeurer tests failing are because of issue #12102 
otherwise, I guess my code is right, I have solved both the problems of issue #12052

### CODE REVIEW

**Comment by tanayag** [comment]
@asmeurer ??

File: sympy/solvers/solvers.py [hunk 1]
```diff
@@ -2746,10 +2746,16 @@ def nsolve(*args, **kwargs):
...
         x0 = args[2]
+        if iterable(fargs)==True and iterable(x0)==True:
+            if len(x0) != len(fargs):
+                raise TypeError('nsolve expected exactly %i guess vectors, got %i'
+                                % (len(fargs), len(x0)))
     elif len(args) == 2:
         f = args[0]
         fargs = None
         x0 = args[1]
+        if iterable(f)==True:
```
Comment (asmeurer) [inline sympy/solvers/solvers.py]: You can just write `if iterable(f)`. The `==True` part is unnecessary.

File: sympy/solvers/solvers.py [hunk 2]
```diff
@@ -2746,10 +2746,16 @@ def nsolve(*args, **kwargs):
         f = args[0]
         fargs = args[1]
         x0 = args[2]
+        if iterable(fargs)==True and iterable(x0)==True:
```
Comment (asmeurer) [inline sympy/solvers/solvers.py]: Same here. `if iterable(fargs) and iterable(x0)`.

**Review by asmeurer — CHANGES_REQUESTED** [review]
This looks good. Can you add tests (use the `raises` function to test that they raise TypeError).

File: sympy/solvers/solvers.py (hunk 2 above)
Comment (tanayag) [inline sympy/solvers/solvers.py]: I'll edit

**Comment by tanayag** [comment]
sure m on it

**Review by asmeurer — APPROVED** [review]

**Comment by tanayag** [comment]
I did a typo
"Ffrom" in first line of solvers_test
but 
corrected it later

**Pull request merged**